In [3]:
import pandas as pd


### Set up
#path = "C:/Users/pterron/Documents/projet_python/Application_ml/Depot_data/"
path = "C:/Users/pierr/Documents/1_projet_python/Valeur_fonciere/Depot_data/Kaggle/"

In [4]:
df_kaggle = pd.read_csv(filepath_or_buffer= path + "submissions.csv", sep=",", decimal=',')

C:\Users\pierr\AppData\Local\Temp\ipykernel_11628\3348054760.py:1: DtypeWarning: Columns (9,11,31) have mixed types. Specify dtype option on import or set low_memory=False.
  df_kaggle = pd.read_csv(filepath_or_buffer= path + "submissions.csv", sep=",", decimal=',')


In [15]:
df_kaggle.shape
#df_kaggle.head()

(378041, 34)

In [16]:
d_type = {
    'Commune' : 'str',
    'Code commune' : 'str',
    'Code departement' : 'str',
    'Code postal' : 'str',
    'Surface reelle bati' : 'float',
    'Nombre pieces principales' : 'float',
    'Surface terrain' : 'float'
}

df = df_kaggle.astype(d_type)
df['Date mutation'] = pd.to_datetime(df['Date mutation'], format="%d/%m/%Y")

df['month'] = df['Date mutation'].dt.month
df['year'] = df['Date mutation'].dt.year



#Garder les colonnes pertinentes
df

Date mutation Nature mutation No voie B/T/Q Type de voie Code voie  \
0         2022-01-03           Vente    13.0   NaN          RUE      2280   
1         2022-01-03           Vente    98.0   NaN          RTE      0055   
2         2022-01-06           Vente   282.0   NaN          RTE      0130   
3         2022-01-05           Vente     7.0   NaN          RUE      0031   
4         2022-01-06           Vente     7.0   NaN          RUE      3125   
...              ...             ...     ...   ...          ...       ...   
378036    2022-06-16    Adjudication    21.0   NaN           AV      6626   
378037    2022-12-14           Vente    13.0   NaN         QUAI      2296   
378038    2022-12-30           Vente    19.0   NaN          RUE      4879   
378039    2022-12-30           Vente    69.0   NaN          RUE      0679   
378040    2022-11-21           Vente    48.0   NaN          RUE      8358   

                  Voie Code postal           Commune Code departement  ...  \
0        DE LA LIBERTE      1000.0   BOURG-EN-BRESSE                1  ...   
1         DE LA DOMBES      1480.0         SAVIGNEUX                1  ...   
2          DE POISATON      1560.0  MANTENAY-MONTLIN                1  ...   
3           DU CORNIER      1150.0     VAUX-EN-BUGEY                1  ...   
4             DES PINS      1000.0   BOURG-EN-BRESSE                1  ...   
...                ...         ...               ...              ...  ...   
378036          MOZART     75016.0          PARIS 16               75  ...   
378037        DE CONTI     75006.0          PARIS 06               75  ...   
378038     JEAN DOLENT     75014.0          PARIS 14               75  ...   
378039        BARRAULT     75013.0          PARIS 13               75  ...   
378040  DE LA ROQUETTE     75011.0          PARIS 11               75  ...   

                                      Type local Identifiant local  \
0                                    Appartement               NaN   
1                                    Appartement               NaN   
2                                         Maison               NaN   
3                                         Maison               NaN   
4                                            NaN               NaN   
...                                          ...               ...   
378036                               Appartement               NaN   
378037                                Dépendance               NaN   
378038  Local industriel. commercial ou assimilé               NaN   
378039                                Dépendance               NaN   
378040                                Dépendance               NaN   

       Surface reelle bati  Nombre pieces principales  Nature culture  \
0                     24.0                        1.0             NaN   
1                    140.0                        3.0             NaN   
2                    108.0                        5.0               S   
3                     85.0                        4.0               S   
4                     99.0                        5.0               S   
...                    ...                        ...             ...   
378036                24.0                        1.0             NaN   
378037                 0.0                        0.0             NaN   
378038               327.0                        0.0             NaN   
378039                 0.0                        0.0             NaN   
378040                 0.0                        0.0             NaN   

       Nature culture speciale  Surface terrain      ID  month  year  
0                          NaN              NaN       1      1  2022  
1                          NaN              NaN       2      1  2022  
2                          NaN            649.0       3      1  2022  
3                          NaN            310.0       4      1  2022  
4                          NaN            765.0       5      1  2022  
.

# latitude longitude

In [27]:
#Import des données
lat = pd.read_csv(filepath_or_buffer= path + "communes-departement-region.csv", sep=",", decimal=".")
#On vire les valeurs qui ne nous interesse pas
lat = lat[['code_postal','latitude', 'longitude','code_commune','code_departement']]
#Suppr valeur vide
lat = lat.dropna(axis=0)

print(lat.shape) #(38932, 5)

#Transtypage
d_type = {
    'code_postal':'float',
    'latitude':'float',
    'longitude':'float',
    'code_commune':'float',
    'code_departement':'str'
}


lat = lat.astype(d_type)
lat['code_commune'] = lat['code_commune'].astype(int).astype(str)
lat['code_postal'] = lat['code_postal'].astype(str)

#Changement des noms
lat = lat.rename(columns={'code_departement': 'Code departement', 'code_commune': 'Code commune', 'code_postal' : 'Code postal'})

lat.drop_duplicates(inplace = True)
print(lat.shape) #(36216, 5)

#Fusion
#Union des deux tables
df = pd.merge(left=df, right=lat, on=['Code departement','Code commune', 'Code postal'], how='left')

print(df.shape) #(378041, 38)

(38932, 5)
(36216, 5)
(378041, 38)


In [28]:
df.head()

Date mutation Nature mutation No voie B/T/Q Type de voie Code voie  \
0    2022-01-03           Vente    13.0   NaN          RUE      2280   
1    2022-01-03           Vente    98.0   NaN          RTE      0055   
2    2022-01-06           Vente   282.0   NaN          RTE      0130   
3    2022-01-05           Vente     7.0   NaN          RUE      0031   
4    2022-01-06           Vente     7.0   NaN          RUE      3125   

            Voie Code postal           Commune Code departement  ...  \
0  DE LA LIBERTE      1000.0   BOURG-EN-BRESSE                1  ...   
1   DE LA DOMBES      1480.0         SAVIGNEUX                1  ...   
2    DE POISATON      1560.0  MANTENAY-MONTLIN                1  ...   
3     DU CORNIER      1150.0     VAUX-EN-BUGEY                1  ...   
4       DES PINS      1000.0   BOURG-EN-BRESSE                1  ...   

  Surface reelle bati Nombre pieces principales Nature culture  \
0                24.0                       1.0            NaN   
1               140.0                       3.0            NaN   
2               108.0                       5.0              S   
3                85.0                       4.0              S   
4                99.0                       5.0              S   

   Nature culture speciale  Surface terrain ID  month  year   latitude  \
0                      NaN              NaN  1      1  2022  46.205152   
1                      NaN              NaN  2      1  2022  45.997347   
2                      NaN            649.0  3      1  2022  46.430275   
3                      NaN            310.0  4      1  2022  45.920121   
4                      NaN            765.0  5      1  2022  46.205152   

  longitude  
0  5.246021  
1  4.857620  
2  5.106759  
3  5.363532  
4  5.246021  

[5 rows x 38 columns]

# Prix au m² (Communes)

In [76]:
#Import des données
prix_m2_com = pd.read_csv(filepath_or_buffer= "C:/Users/pierr/Documents/1_projet_python/Valeur_fonciere/Depot_data/prix_m2_com_region.csv", sep=",", decimal=".")
prix_m2_com['Code commune'] = prix_m2_com['Code commune'].astype(str)

In [80]:
prix_m2_com.value_counts()

Commune                Code departement  Code commune  Prix m2 moyen region  Prix m2 moyen commune
LAMPAUL-PLOUDALMEZEAU  29                99            1895.382583           1833.861041              2
ST-JEAN-DU-DOIGT       29                251           1895.382583           1765.085621              2
FRONTIGNAN             34                108           2772.260140           3098.148698              2
SAINT-URBAIN           29                270           1895.382583           2005.622713              2
TREBABU                29                282           1895.382583           2369.670268              2
                                                                                                     ..
GARIN                  31                213           2886.356271           1066.780303              1
GARIGNY                18                99            1132.497445           750.300394               1
GARIDECH               31                212           2886.356271   

In [83]:
prix_m2_com.drop_duplicates(inplace= True)

In [84]:
prix_m2_com.info()
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 27632 entries, 0 to 27809
Data columns (total 5 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Commune                27632 non-null  object 
 1   Code departement       27632 non-null  object 
 2   Code commune           27632 non-null  object 
 3   Prix m2 moyen region   27632 non-null  float64
 4   Prix m2 moyen commune  27632 non-null  float64
dtypes: float64(2), object(3)
memory usage: 1.3+ MB
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 378041 entries, 0 to 378040
Data columns (total 38 columns):
 #   Column                      Non-Null Count   Dtype         
---  ------                      --------------   -----         
 0   Date mutation               378041 non-null  datetime64[ns]
 1   Nature mutation             378041 non-null  object        
 2   No voie                     375777 non-null  object        
 3   B/T/Q                       26339 non-nu

In [85]:
df_test = pd.merge(left=df, right=prix_m2_com[['Code departement','Code commune','Prix m2 moyen commune']], on=['Code departement','Code commune'], how='left')

In [86]:
df_test['Prix m2 moyen commune'].isna().sum() #3543

3543

In [87]:
df_test.shape #(378041, 39)

(378041, 39)

In [90]:
df_prix_dep = prix_m2_com[['Code departement','Prix m2 moyen region']].drop_duplicates()

In [94]:
df_prix_dep

Code departement  Prix m2 moyen region
0                    1           2315.872738
366                  2           1276.944087
962                  3           1131.842581
1234                 4           2031.920547
1399                 5           2369.611855
...                ...                   ...
27687              971           2566.968542
27720              972           2299.980885
27753              973           2293.808491
27766              974           2476.421943
27790               75          10069.780223

[97 rows x 2 columns]

In [95]:
df_test = pd.merge(left=df_test, right=df_prix_dep, on='Code departement', how='left')

# Remplacement valeurs vides

In [100]:
df_fin = df_test[['Type local','Surface reelle bati','Nombre pieces principales','year','latitude','longitude','Prix m2 moyen region','Prix m2 moyen commune']]

In [104]:
df_fin.shape

(378041, 8)

In [105]:
df_fin.isna().sum()

Type local                   50000
Surface reelle bati            541
Nombre pieces principales      541
year                             0
latitude                      1317
longitude                     1317
Prix m2 moyen region             0
Prix m2 moyen commune         3543
dtype: int64

In [109]:
#Remplacement des vlauers vides
df_fin['Surface reelle bati'] = df_fin['Surface reelle bati'].fillna(value = df_fin['Surface reelle bati'].median())
df_fin['Nombre pieces principales'] = df_fin['Nombre pieces principales'].fillna(value = 4)
#On remplace les valeurs nulles lat et longi par zero
values = {"latitude": 0, "longitude":0}
df_fin = df_fin.fillna(value = values)

df_fin['Prix m2 moyen commune'].fillna(df_fin['Prix m2 moyen region'], inplace= True)

In [110]:
df_fin.isna().sum()

Type local                   50000
Surface reelle bati              0
Nombre pieces principales        0
year                             0
latitude                         0
longitude                        0
Prix m2 moyen region             0
Prix m2 moyen commune            0
dtype: int64

In [116]:
#df_fin[df_fin['Prix m2 moyen commune'] == df_fin['Prix m2 moyen region']]
df_fin.shape

(378041, 8)

In [117]:
import pickle

#Extract
X_missing = df_fin[df_fin['Type local'].isna()]

X_missing

Type local  Surface reelle bati  Nombre pieces principales  year  \
4             NaN                 99.0                        5.0  2022   
13            NaN                 76.0                        3.0  2022   
33            NaN                 95.0                        5.0  2022   
40            NaN                230.0                        0.0  2022   
44            NaN                  0.0                        0.0  2022   
...           ...                  ...                        ...   ...   
378010        NaN                 34.0                        2.0  2022   
378015        NaN                  0.0                        0.0  2022   
378019        NaN                  0.0                        0.0  2022   
378023        NaN                 10.0                        1.0  2022   
378024        NaN                 10.0                        0.0  2022   

         latitude  longitude  Prix m2 moyen region  Prix m2 moyen commune  
4       46.205152   5.246021           2315.872738            2509.869124  
13      46.364383   4.902428           2315.872738            1662.842491  
33      46.299763   6.100580           2315.872738            4039.331560  
40      45.844210   4.941067           2315.872738            3525.708737  
44      46.254539   6.036291           2315.872738            4610.464963  
...           ...        ...                   ...                    ...  
378010  48.859415   2.378741          10069.780223           10422.853766  
378015  48.859415   2.378741          10069.780223           10422.853766  
378019  48.860399   2.262100          10069.780223           10293.521223  
378023  48.828718   2.362468          10069.780223            9286.449714  
378024  48.835156   2.419807          10069.780223            9922.648119  

[50000 rows x 8 columns]

In [118]:

#On importe le modèle pickle pour le type local
pickled_model = pickle.load(open('treemodel_local.pkl', 'rb'))
X_missing['Type local'] = pickled_model.predict(X_missing[['Surface reelle bati', 'Prix m2 moyen commune']])

C:\Users\pierr\AppData\Local\Temp\ipykernel_11628\1520109459.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  X_missing['Type local'] = pickled_model.predict(X_missing[['Surface reelle bati', 'Prix m2 moyen commune']])


In [119]:
X_missing.index

Index([     4,     13,     33,     40,     44,     46,     54,     56,     70,
           88,
       ...
       377951, 377960, 377967, 377974, 377986, 378010, 378015, 378019, 378023,
       378024],
      dtype='int64', length=50000)

In [121]:
X_missing['Type local'].value_counts()

Type local
Maison         35439
Appartement    14561
Name: count, dtype: int64

In [122]:
df_fin.loc[X_missing.index,'Type local'] = X_missing['Type local']

In [123]:
df_fin.iloc[X_missing.index]['Type local']

4              Maison
13             Maison
33             Maison
40             Maison
44        Appartement
             ...     
378010    Appartement
378015    Appartement
378019    Appartement
378023    Appartement
378024    Appartement
Name: Type local, Length: 50000, dtype: object

In [82]:

#df_fin['Surface terrain'].loc[df_fin['Surface terrain'].isna() & (df_fin['Type local'] == 'Appartement')] = 0
#df_fin['Surface terrain'] = df_fin['Surface terrain'].fillna(value = df_fin['Surface terrain'].median())

C:\Users\pterron\AppData\Local\Temp\ipykernel_3544\3852283746.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_fin['Surface terrain'].loc[df_fin['Surface terrain'].isna() & (df_fin['Type local'] == 'Appartement')] = 0
C:\Users\pterron\AppData\Local\Temp\ipykernel_3544\3852283746.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_fin['Surface terrain'] = df_fin['Surface terrain'].fillna(value = df_fin['Surface terrain'].median())


In [124]:
df_fin.isna().sum()/len(df_fin)

Type local                   0.0
Surface reelle bati          0.0
Nombre pieces principales    0.0
year                         0.0
latitude                     0.0
longitude                    0.0
Prix m2 moyen region         0.0
Prix m2 moyen commune        0.0
dtype: float64

In [169]:
df_fin.shape
df_fin['Type local'].value_counts()

Type local
Maison                                      228864
Appartement                                  76133
Dépendance                                   46331
Local industriel. commercial ou assimilé     26713
Name: count, dtype: int64

In [160]:
#On importe le modèle pickle pour le prix
prix_model_maison = pickle.load(open('reg_prix_maison.pkl', 'rb'))
prix_model_appart = pickle.load(open('reg_prix_appart.pkl', 'rb'))

df_maison = df_fin[df_fin['Type local'] == 'Maison'][['Surface reelle bati', 'Prix m2 moyen commune']]
df_maison

Surface reelle bati  Prix m2 moyen commune
2                     108.0            3095.238095
3                      85.0            1972.221880
4                      99.0            2509.869124
7                      82.0            1716.822618
8                     180.0            1745.731327
...                     ...                    ...
377873                168.0           10503.884439
377943                182.0           10293.521223
377960                 88.0           10107.192802
377967                115.0           10293.521223
377974                150.0           10293.521223

[228864 rows x 2 columns]

In [191]:
df_fin_test = df_fin

df_maison = df_fin[df_fin['Type local'] == 'Maison'][['Surface reelle bati', 'Prix m2 moyen commune']]
df_appart = df_fin[df_fin['Type local'] == 'Appartement'][['Surface reelle bati', 'Prix m2 moyen commune']]
df_locaux = df_fin[df_fin['Type local'] == 'Local industriel. commercial ou assimilé'][['Surface reelle bati','Prix m2 moyen region']]
df_dep = df_fin[df_fin['Type local'] == 'Dépendance']

df_fin_test.loc[df_maison.index,'y_pred'] = prix_model_maison.predict(df_maison)
df_fin_test.loc[df_appart.index,'y_pred'] = prix_model_appart.predict(df_appart)
df_fin_test.loc[df_locaux.index,'y_pred'] = df_locaux['Surface reelle bati'] * df_locaux['Prix m2 moyen region']

df_fin_test.loc[df_dep.index,'y_pred'] = 40000

df_fin_test[df_fin_test['y_pred'] == 0] = 98000

In [200]:
df_fin[(df_fin['Type local'] == 'Local industriel. commercial ou assimilé') & ()]

Type local  Surface reelle bati  \
19      Local industriel. commercial ou assimilé                286.0   
20      Local industriel. commercial ou assimilé                 75.0   
52      Local industriel. commercial ou assimilé                382.0   
61      Local industriel. commercial ou assimilé               2100.0   
86      Local industriel. commercial ou assimilé                875.0   
...                                          ...                  ...   
378025  Local industriel. commercial ou assimilé                155.0   
378026  Local industriel. commercial ou assimilé                 15.0   
378029  Local industriel. commercial ou assimilé                 53.0   
378032  Local industriel. commercial ou assimilé                428.0   
378038  Local industriel. commercial ou assimilé                327.0   

        Nombre pieces principales  year   latitude  longitude  \
19                            0.0  2022  46.205152   5.246021   
20                            4.0  2022  46.226004   5.610954   
52                            0.0  2022  46.205152   5.246021   
61                            0.0  2022  46.325281   5.112960   
86                            0.0  2022  45.844210   4.941067   
...                           ...   ...        ...        ...   
378025                        0.0  2022  48.835156   2.419807   
378026                        0.0  2022  48.828993   2.327101   
378029                        0.0  2022  48.835156   2.419807   
378032                        0.0  2022  48.848968   2.332671   
378038                        0.0  2022  48.828993   2.327101   

        Prix m2 moyen region  Prix m2 moyen commune        y_pred  
19               2315.872738            2509.869124  6.623396e+05  
20               2315.872738            2026.838886  1.736905e+05  
52               2315.872738            2509.869124  8.846634e+05  
61               2315.872738            1743.290829  4.863333e+06  
86               2315.872738            3525.708737  2.026389e+06  
...                      ...                    ...           ...  
378025          10069.780223            9922.648119  1.560816e+06  
378026          10069.780223           10358.062737  1.510467e+05  
378029          10069.780223            9922.648119  5.336984e+05  
378032          10069.780223           12965.121910  4.309866e+06  
378038          10069.780223           10358.062737  3.292818e+06  

[26059 rows x 9 columns]

In [199]:
df_fin_test.sort_values(by='y_pred')

Type local  Surface reelle bati  \
23573   Local industriel. commercial ou assimilé                  1.0   
185675  Local industriel. commercial ou assimilé                  1.0   
185080  Local industriel. commercial ou assimilé                  1.0   
63742   Local industriel. commercial ou assimilé                  5.0   
260289  Local industriel. commercial ou assimilé                  2.0   
...                                          ...                  ...   
276998  Local industriel. commercial ou assimilé              30158.0   
237487  Local industriel. commercial ou assimilé              36960.0   
153324  Local industriel. commercial ou assimilé              46945.0   
357873  Local industriel. commercial ou assimilé              41970.0   
255086  Local industriel. commercial ou assimilé              48183.0   

        Nombre pieces principales  year   latitude  longitude  \
23573                         0.0  2022  49.748099   4.685674   
185675                        0.0  2022  47.697349  -2.830809   
185080                        0.0  2022  47.663802  -2.436711   
63742                         0.0  2022  46.368093   2.215611   
260289                        0.0  2022  49.441346   1.092568   
...                           ...   ...        ...        ...   
276998                        0.0  2022  48.962692   1.848703   
237487                        0.0  2022  45.987460   4.731740   
153324                        0.0  2022  47.360680  -1.617545   
357873                        0.0  2022  49.044168   2.128613   
255086                        0.0  2022  45.633432   6.874624   

        Prix m2 moyen region  Prix m2 moyen commune        y_pred  
23573            1234.333333            1877.945047  1.234333e+03  
185675           2353.124065            2709.743175  2.353124e+03  
185080           2353.124065            2227.689693  2.353124e+03  
63742             736.889653             791.990031  3.684448e+03  
260289           2152.789575            2887.761425  4.305579e+03  
...                      ...                    ...           ...  
276998           3898.308462            3088.555368  1.175652e+08  
237487           3780.174023            2330.035687  1.397152e+08  
153324           3057.976614            2920.098830  1.435567e+08  
357873           3628.241928            3153.853257  1.522773e+08  
255086           3325.708068            5355.733521  1.602426e+08  

[378041 rows x 9 columns]

In [202]:
print('Local')
print(df_fin_test[df_fin_test['Type local'] == 'Local industriel. commercial ou assimilé']['y_pred'].describe().apply(lambda x: format(x, 'f')))
print('maison')
print(df_fin_test[df_fin_test['Type local'] == 'Maison']['y_pred'].describe().apply(lambda x: format(x, 'f')))
print('Appartement')
print(df_fin_test[df_fin_test['Type local'] == 'Appartement']['y_pred'].describe().apply(lambda x: format(x, 'f')))
print('Dépendance')
print(df_fin_test[df_fin_test['Type local'] == 'Dépendance']['y_pred'].describe().apply(lambda x: format(x, 'f')))

Local
count        26059.000000
mean       1190630.744271
std        4478480.359870
min           1234.333333
25%         128091.525011
50%         287908.456641
75%         767765.843923
max      160242591.852288
Name: y_pred, dtype: object
maison
count    228864.000000
mean     206132.562977
std       95480.583179
min       35714.976947
25%      130079.006351
50%      195321.008412
75%      282730.949289
max      382238.854369
Name: y_pred, dtype: object
Appartement
count     76133.000000
mean     161467.966199
std       84356.117694
min       37921.350396
25%       96549.344678
50%      142975.842209
75%      208461.583502
max      360953.564418
Name: y_pred, dtype: object
Dépendance
count    46331.000000
mean     40000.000000
std          0.000000
min      40000.000000
25%      40000.000000
50%      40000.000000
75%      40000.000000
max      40000.000000
Name: y_pred, dtype: object


count       378041.000000
mean        244453.226313
std        1207724.306522
min           1234.333333
25%          89219.014116
50%         160657.707663
75%         259491.888280
max      160242591.852288
Name: y_pred, dtype: float64
count    418379.000000
mean     177645.918172
std      108670.010649
min           0.000000
25%       93000.000000
50%      160000.000000
75%      244000.000000
max      498000.000000

In [144]:
prix_model_maison.predict(df_fin_test[df_fin_test['Type local'] == 'Maison'][['Surface reelle bati', 'Prix m2 moyen commune']]).size

228864

In [86]:
#Y_pred en fonction du type del ocal
for i in df_fin.index:
    if(df_fin)

Type local  Surface reelle bati  Nombre pieces principales  \
0       Appartement                 24.0                        1.0   
1       Appartement                140.0                        3.0   
18      Appartement                 25.0                        1.0   
32      Appartement                 78.0                        3.0   
54      Appartement                 43.0                        2.0   
...             ...                  ...                        ...   
378030  Appartement                 56.0                        2.0   
378031  Appartement                 34.0                        2.0   
378033  Appartement                 13.0                        1.0   
378034  Appartement                 97.0                        3.0   
378036  Appartement                 24.0                        1.0   

        Surface terrain  month  year  
0                   0.0      1  2022  
1                   0.0      1  2022  
18                  0.0      1  2022  
32                  0.0      1  2022  
54                  0.0      1  2022  
...                 ...    ...   ...  
378030              0.0     12  2022  
378031              0.0     12  2022  
378033              0.0     12  2022  
378034              0.0     12  2022  
378036              0.0      6  2022  

[74906 rows x 6 columns]